js/algoliasearch/click_conversion_analytics.js

```diff
@@ -1,89 +1,108 @@
 algoliaBundle.$(function ($) {
 	AlgoliaAnalytics.init({
-		applicationID: algoliaConfig.applicationId,
+		appId: algoliaConfig.applicationId,
 		apiKey: algoliaConfig.instant.apiKey
 	});
 
 	// "Click" in autocomplete
 	$(algoliaConfig.autocomplete.selector).each(function () {
-		$(this).on('autocomplete:selected', function (e, suggestion) {
-			trackClick(suggestion.objectID, suggestion.__position, suggestion.__queryID);
+		$(this).on('autocomplete:selected', function (e, suggestion, dataset, context) {
+
+			var sources = analyticsHelper.sources;
+			var source = sources.find(function(e) {
+				return e.name == dataset;
+			});
+
+			trackClick(source.indexName, suggestion.objectID, suggestion.__position, suggestion.__queryID);
```

Comment:
source is the result of `find`, so it can be undefined. It's best to check before trying to access .indexName

Reply:
I've switched it to use `filter` instead which should return an array. We can check if filter returns a count before accessing the source.